packages/kotti-ui/source/kotti-field-file-upload/components/DropArea.vue

```diff
@@ -0,0 +1,265 @@
+				v-text="translations.text.learnMore"
+			/>
+		</div>
+		<input
+			v-show="false"
+			ref="uploadInputRef"
+			:accept="accept"
+			:data-test="dataTest ? `${dataTest}.input` : undefined"
+			:multiple="allowMultiple"
+			type="file"
+			@change="onSelect"
+			@click.stop
+		/>
+		<div
+			v-if="$slots.footer"
+			class="kt-field-file-upload__drop-area__footer-slot"
+			@mouseenter.prevent.stop="isHovering = false"
+			@mouseleave.prevent.stop="isHovering = true"
+		>
+			<slot name="footer" />
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import {
+	computed,
+	defineComponent,
+	onBeforeMount,
+	onUnmounted,
+	ref,
+} from '@vue/composition-api'
+
+import { useTranslationNamespace } from '../../kotti-i18n/hooks'
+import { KEY_ENTER, KEY_SPACE } from '../constants'
+import { KottiFieldFileUpload } from '../types'
+import { formatAccept, formatFileSize, getFileItem } from '../utils'
+import { validateSelection } from '../validators'
+
+export default defineComponent({
+	name: 'DropArea',
+	props: {
+		accept: { default: null, type: String },
+		acceptLabel: { default: null, type: String },
+		allowMultiple: { default: true, type: Boolean },
+		dataTest: { default: null, type: String },
+		icon: { default: null, type: String },
+		infoUrl: { default: null, type: String },
+		isDisabled: { default: false, type: Boolean },
+		maxFileSize: { default: null, type: Number },
+		tabIndex: { default: 0, type: Number },
+	},
+	setup(
+		props: Pick<
+			KottiFieldFileUpload.PropsInternal,
+			| 'accept'
+			| 'acceptLabel'
+			| 'allowMultiple'
+			| 'dataTest'
+			| 'icon'
+			| 'infoUrl'
+			| 'isDisabled'
+			| 'maxFileSize'
+			| 'tabIndex'
+		>,
+		{ emit },
+	) {
+		const translations = useTranslationNamespace('KtFieldFileUpload')
+
+		const dropAreaRef = ref<HTMLDivElement | null>(null)
+		const isDragging = ref<boolean>(false)
+		const isHovering = ref<boolean>(false)
+		const uploadInputRef = ref<HTMLInputElement | null>(null)
+
+		const emitFiles = (files: FileList | null) => {
+			if (props.isDisabled || !files || files.length === 0) return
+
+			const { accept, allowMultiple, maxFileSize } = props
+			const validation = validateSelection({
+				accept,
+				allowMultiple,
+				fileName: files[0].name,
+				fileSize: files[0].size,
+				maxFileSize,
+				selection: files.length,
+			})
+
+			if (validation !== KottiFieldFileUpload.Validation.SUCCESS) {
+				emit('error', translations.value.validationMsg[validation])
+				return
+			}
+
+			emit('addFiles', Array.from(files).map(getFileItem))
+		}
+
+		const dropAreaListener = (event: KeyboardEvent) => {
+			if (
+				!props.isDisabled &&
+				[KEY_ENTER, KEY_SPACE].includes(event.key) &&
+				dropAreaRef.value === window.document.activeElement
+			) {
+				event.preventDefault()
+				event.stopPropagation()
+				uploadInputRef.value?.click()
+			}
+		}
+
+		onBeforeMount(() => window.addEventListener('keydown', dropAreaListener))
```

Comment:
I'm not sure if performance-wise it'd be more optimal but we can research 

But I'd personally add the event listener specifically to the keydown.KEY_ENTER, and another for the keydown.SPACE

imagine the amount of times this would get called within a window

technically it's very uncosty because you v-if but **MAYBE** performance-wise it's better to directly watch on the specific event

Reply:
also, out of curiosity

did you try just listening on the element with the ref: `uploadInputRef` directly


```
ref="uploadInputRef"
@keydown.space.stop.prevent="handleDropAreaFocus"
@keydown.enter.stop.prevent="handleDropAreaFocus"
```

Reply:
Sorry - I'm confused :D

the @click on this element is @click.stop without a handler

why do you programatically trigger a click?

Reply:
Added `@keydown.space` and `@keydown.enter` instead of the listener but they need to go on the DIV container and not the input because it's hidden (intended)

The problem was the label that captures all clicks inside including clicks on the items bellow the drop area. If you remember it also messes up with the focus of the buttons inside. 

Anyway did some refactor, it should be good now

I still need to trigger the input click event on the DIV keydown events to emulate the focus behavior